Forge.TreeWalker.UnitTests/test/TreeSchemaValidatorTests.cs

```diff
@@ -0,0 +1,175 @@
+            bool res = ForgeSchemaValidator.ValidateSchemaAsForgeTreeDictionary(treeSchemas, linkedRulesForDictionary, true, out IList<ValidationError> errorList);
+            Assert.IsTrue(res);
+            Assert.AreEqual(0, errorList.Count);
+        }
+
+        [TestMethod]
+        public void Test_ValidateSchemaAsString_WithForgeTree()
+        {
+            bool res = ForgeSchemaValidator.ValidateSchemaAsString(forgeTreeAsString, stringRules, false, out IList<ValidationError> errorList);
+            Assert.IsTrue(res);
+            Assert.AreEqual(0, errorList.Count);
+        }
+
+        [TestMethod]
+        public void Test_ValidateSchemaAsString_WithValidateAsDictionary()
+        {
+            bool res = ForgeSchemaValidator.ValidateSchemaAsString(forgeTreeDictionaryAsString, linkedRulesForDictionary, true, out IList<ValidationError> errorList);
+            Assert.IsTrue(res);
+            Assert.AreEqual(0, errorList.Count);
+        }
+
+        [TestMethod]
+        public void Test_ValidateSchemaFromPath_WithoutValidateAsDictionary()
+        {
+            bool res = ForgeSchemaValidator.ValidateSchemaFromPath(forgeTreeFromPath, stringRules, false, out IList<ValidationError> errorList);
+            Assert.IsTrue(res);
+            Assert.AreEqual(0, errorList.Count);
+        }
+
+        [TestMethod]
+        public void Test_ValidateSchemasFromDirectory_WithValidateAsSeparateFiles()
+        {
+            bool res = ForgeSchemaValidator.ValidateSchemaFromDirectory(schemaDirectoryPath, stringRules, false, out IList<ValidationError> errorList);
+            Assert.IsTrue(res);
+            Assert.AreEqual(0, errorList.Count);
+        }
+
+        [TestMethod]
+        public void Test_GetLinkedJSchemaRules_WithInvalidUrl__Fail()
+        {
+            Assert.ThrowsException<JSchemaReaderException>(() => ForgeSchemaValidator.GetLinkedJSchemaRules(rulesForDictionary, stringRules, "//nameImadeup.json"));
+        }
+
+        [TestMethod]
+        public void Test_ValidateSchemaAsForgeTree_WithEmptyStringRules__Fail()
+        {
+            Assert.ThrowsException<ArgumentException>(() => ForgeSchemaValidator.ValidateSchemaAsForgeTree(treeSchema, "", out IList<ValidationError> errorList));
+        }
+
+        [TestMethod]
+        public void Test_ValidateSchemaAsString_WithErrorContent_Fail()
+        {
+            bool res = ForgeSchemaValidator.ValidateSchemaAsString(invalidSchemaWithErrorContent, jschemaRules, false, out IList<ValidationError> errorList);
+            Assert.AreEqual(false, res);
+            Assert.AreEqual("JSON is valid against no schemas from 'oneOf'.", errorList.First().Message);
+        }
+
+        [TestMethod]
+        public void Test_ValidateSchema_FromInvalidDirectoryPath_Fail()
+        {
+            Assert.ThrowsException<IOException>(() => ForgeSchemaValidator.ValidateSchemaFromDirectory(invalidSchemaDirectoryPath, jschemaRules, false, out IList<ValidationError> errorList));
```

Comment:
> Assert.ThrowsException<IOException>(() [](http://example.com/codeflow?start=11&length=39)

These last 3 tests are getting a little long. Typically we don't want a line of code to go beyond 150 columns.

`
Assert.ThrowsException<NullReferenceException>(
    () => ForgeSchemaValidator.ValidateSchemaFromDirectory(
                schemaDirectoryPath,
                linkedRulesForDictionary,
                true,
                out IList<ValidationError> errorList));
`